Sources/UIScrollView.swift

```diff
@@ -23,7 +23,13 @@ open class UIScrollView: UIView {
         get { return bounds.origin }
         set {
             guard newValue != contentOffset else { return }
-            cancelDecelerationAnimations()
+
+            // Cancel deceleration animations only when contentOffset gets set without animations.
+            // Otherwise we might cancel any "bounds" animations which are not iniated from velocity scrolling.
```

Comment:
a couple of things:
- I'm not a fan of making `transactionStack` more accessible than `private` as it was (I don't want other parts of the code relying on it - especially not something as high level as `UIScrollView`; it will lead to spaghetti code and weird dependencies that are hard to debug)
- After looking at this code and the comments for quite a while I still can't figure out what the "transaction stack is empty" thing is doing in the first place. Can we look to see whether the scrollview's layer has a `bounds` animation on it instead?